### ISSUE TITLE
32490629: Getting today's date in YYYY-MM-DD in Python

### ISSUE THREAD

**Issue by arungudelli** [arungudelli/python.shiksha#92]
## Description

Write a detailed tutorial about [Getting today's date in YYYY-MM-DD in Python?](https://stackoverflow.com/questions/32490629/getting-todays-date-in-yyyy-mm-dd-in-python)

## Stackoverflow link

Write different ways to solve the problem

https://stackoverflow.com/questions/32490629/getting-todays-date-in-yyyy-mm-dd-in-python

## Read the contributing guidelines

https://github.com/arungudelli/python.shiksha/blob/f2e70e57e90988b3db839041a472ea6db3a1b262/CONTRIBUTING.md

## Have a look at the sample article.

https://python.shiksha/tips/check-substring-in-string_python/

## File format:

https://github.com/arungudelli/python.shiksha/blob/main/content/en/tips/check-substring-in-string_python.md

**Comment by arungudelli** [comment]
@AbhignyaTayi Please work on this..

**Comment by AbhignyaTayi** [comment]
###  Getting today's date in YYYY-MM-DD in Python

This article will discuss different ways of getting today's data in the YYYY-MM-DD format using Python. We will cover the two most popular methods and explain how they work. Also, we will talk about any method-specific features for the Python language version.

**Getting Today's Date in YYYY-MM-DD Format in Python:**

The four most popular methods to get today's date in the YYYY-MM-DD format using Python are the `datetime` module , `strftime()` method, the `arrow `module, and the `time` module.

**1. Using the `datetime` module:**

The `datetime` module is a built-in module in Python that provides classes for working with dates and times. We can use the `datetime.today()` method to get today's date and then use the `strftime() `method to format it to the YYYY-MM-DD format.

Code : 

```
import datetime

today = datetime.datetime.today().strftime('%Y-%m-%d')

print(today)
```

Output:

```
2023-02-15

```

**2. Using the `strftime()` method:**

We can also get today's date in the YYYY-MM-DD format using the strftime() method. The strftime() method is used to format date and time objects in Python. We can pass the format string '%Y-%m-%d' to the strftime() method to get today's date in the YYYY-MM-DD format.

Code : 

```
import time

today = time.strftime('%Y-%m-%d')

print(today)

```

Output : 

```
2023-02-15

```

**3. Using the `arrow` module:**

The arrow module is a third-party library that provides a more human-friendly API for working with dates and times. It is not included in the Python standard library, so you will need to install it separately using pip.

Code : 

```
import arrow

today = arrow.now()
formatted_date = today.format('YYYY-MM-DD')

print("Today's date in YYYY-MM-DD format:", formatted_date)


```
Output : 

```
Today's date in YYYY-MM-DD format: 2023-02-15

```

**4. Using the time module**

Python's built-in time module provides a localtime() function that we can use to get the current time as a struct_time object. We can then extract the year, month, and day from the struct_time object and format them as a string in YYYY-MM-DD format.

Code : 

```
import time

current_time = time.localtime()
formatted_date = time.strftime("%Y-%m-%d", current_time)

print("Today's date in YYYY-MM-DD format:", formatted_date)

```

Output : 

```
Today's date in YYYY-MM-DD format: 2023-02-15

```

**Method-specific Features**

Both the `datetime` module and the `strftime()` method are available in all Python versions. However, the `datetime` module is only available in Python 2.4 and later versions.

**Conclusion**

In conclusion, there are several ways to get today's date in the format of YYYY-MM-DD in Python. The four methods we have covered are the `Arrow` library, the `datetime` module, the `time `module, and the `strftime` function.

The `Arrow` library is a powerful and flexible library for working with dates and times. It provides a simple and intuitive way to work with dates and times and makes it easy to format them in various ways.

The `datetime` module is a built-in module in Python that provides various classes for working with dates and times. It is a standard way to work with dates and times in Python and provides many useful features for working with them.

The `time` module is another built-in module in Python that provides functions for working with time-related data. It is useful for getting the current date and time, and for working with timestamps.

The `strftime` function is a method for formatting dates and times in Python. It provides a way to convert dates and times to string format and allows you to specify the format you want to use.

Overall, there are multiple ways to get today's date in Python, and the choice of method depends on the specific requirements of your project. By understanding these methods, you can choose the one that best suits your needs and work efficiently with date and time data in Python.
